### ISSUE TITLE
Itinéraire qui sort du département testé 🥐

### ISSUE THREAD

**Issue by mquandalle** [laem/villes.plus#142]
La [cyclabilité du département des Hauts-de-Seine](https://www.villes.plus/cyclables/Hauts-de-Seine) teste des itinéraires parisiens : 

![image](https://github.com/laem/villes.plus/assets/1730702/1d2eccf6-43dd-4935-be80-3afee2d66040)

Lorsqu'on a un territoire en forme de croissant, ou plus généralement toute forme non-convexe, un segment entre 2 points du territoire peut en sortir avant d'y revenir.

**Comment by laem** [comment]
Oui, bien vu, je l'avais déjà découvert dans une autre issue. Ça me semble : 
- techniquement difficile de dire au routeur de rester dans la zone
- politiquement plutôt intéressant de montrer que certains territoires sont en dépendance d'un autre, et donc dépendent des avancées de l'autre, car c'est la réalité du cycliste qui se déplace de vers son territoire. 

Qu'en penses-tu ?

**Comment by mquandalle** [comment]
Pour le problème n'est pas que ça déborde un peu, mais d'avoir de grands trajets hors du département, qui donc ont une influence statistique, comme les grands trajets dans Paris ci-dessus.

En l'occurence il me semble qu'il ne faudrait pas tracer d'itinéraire entre Montrouge et Villeneuve-la-Garenne (à l'Est de Saint Denis). Ça me semble lié à l’algo qui prend les points les plus proches dans chaque direction, mais dans un cas de croissant comme ici, le point le plus proche sur un axe Nord-Sud fait traverser tout Paris de manière inattendue.

**Comment by mquandalle** [comment]
Idée en l'air d'algo : calculer l'enveloppe convexe ([algos wokipedia](https://fr.wikipedia.org/wiki/Enveloppe_convexe#Aspects_algorithmiques)) et pour les points à la frontière, calculer seulement les directions vers l'intérieur de la forme.

**Comment by laem** [comment]
Ah oui, en effet intéressant ! Peut-être aussi simplement réduire la distance de test dans une direction. À creuser. Je note même si ça me semble pas prioritaire, car ça concerne assez peu de départements.

**Comment by mquandalle** [comment]
Oui j'avais pensé à la distance aussi. Ça me semble moins robuste s'il y a effectivement une longue distance à parcourir dans une direction tout en restant sur le territoire testé (un grand parc sans mairies ni arrêts de bus par exemple)
